include/lp_hash_table.h

```diff
@@ -0,0 +1,113 @@
+#ifndef DIL_LP_HASH_TABLE_H
+#define DIL_LP_HASH_TABLE_H
+
+#include <assert.h>
+#include <functional>
+#include <stdint.h>
+#include <stdlib.h>
+#include <type_traits>
+
+/*
+ * LpHashTable is a map of key-value pairs where keys are constant strings
```

Comment:
Ok, that's much better! :)
Please use "keys are `const char *`" just to not imply that we save strings in any way.
(I know you mention it below and that's good, but also people when trying to write code fast may not look on the notes initially and so one sentence can confuse them).

Reply:
It's funny because that's how I originally wrote it down but then I thought that it might show too much internal implementation haha.

Reply:
Well.. The user has to know that it saves `const char *` 'cause they have to pass one. :P 
As a general thought, it's kind of "artistic" how much info you give the user.
The convention is: "Abstract as much as you can from the user". That gets worse with all the OOP, getters, setters and the like.
I'm not a fan of it. I don't think the user should think they use some kind of magic.
Knowing some info about what the thing does means you can e.g. `reserve()` in a `std::vector` and save a lot of time.

Reply:
I agree

Reply:
And btw I forgot: You may now write the doc for the internals :)